parsec/parsec_mpi_funnelled.c

```diff
@@ -726,8 +726,11 @@ mpi_no_thread_tag_unregister(parsec_ce_tag_t tag)
     MPI_Status status;
 
     for(i = tag_struct->start_idx; i < tag_struct->start_idx + EACH_STATIC_REQ_RANGE; i++) {
+#if !defined(CRAY_MPICH_VERSION)
+        // MPI Cancel broken on Cray
         MPI_Cancel(&array_of_requests[i]);
         MPI_Test(&array_of_requests[i], &flag, &status);
+#endif
```

Comment:
So in their world it is legitimate to free a pending request but not to cancel it ? Is this even compliant with the MPI standard ?

Reply:
Yes and no. Yes, it is valid to FREE a request that is pending without first cancelling it. No it is not valid to enter FINALIZE with a freed request that has no semantic garantees of being completable (i.e., all the required SEND to satisfy the FREEd RECV requests are posted before calling FINALIZE, and vice-versa). 

Open MPI would error (or at the minimum issue some warning) when during finalize we find some uncompleted/uncompletable requests. MPICH doesn't. On the other hand OMPI doesn't crash when calling CANCEL on a manifestly not-started request.